mlx/backend/common/inverse.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright © 2023-2024 Apple Inc.
+
+#include "mlx/allocator.h"
+#include "mlx/backend/common/copy.h"
+#include "mlx/primitives.h"
+
+#ifdef ACCELERATE_NEW_LAPACK
+#include <Accelerate/Accelerate.h>
+#else
+#include <lapack.h>
+#endif
+
+namespace mlx::core {
+
+void inverse_impl(const array& a, array& inv) {
+  // Lapack uses the column-major convention. We take advantage of the following
+  // identity to avoid transposing (see
+  // https://math.stackexchange.com/a/340234):
+  //   (A⁻¹)ᵀ = (Aᵀ)⁻¹
+
+  // The inverse is computed in place, so just copy the input to the output.
+  copy(a, inv, a.flags().row_contiguous ? CopyType::Vector : CopyType::General);
+
+  const int N = a.shape(-1);
+  const size_t num_matrices = a.size() / (N * N);
+
+  int info;
+  std::vector<int> ipiv;
+  ipiv.resize(N);
```

Comment:
I noticed you did this in SVD and on second thought, I think it would be better to not use a std::vector for this pattern. It kind of subverts our allocator and so we wouldn't get a good read on e.g. peak memory use (or able to cache the allocation for repeat applications).

Reply:
Maybe instead use a scratch allocation

Reply:
Yes, of course! I just did like it this for RAII, but now that you showed `Buffer::Data`, I'll just use that instead.